graylog2-server/src/main/java/org/graylog2/security/realm/LdapUserAuthenticator.java

```diff
@@ -159,11 +141,77 @@ protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken authtok
         return null;
     }
 
+    protected LdapNetworkConnection openLdapConnection(LdapSettings ldapSettings) throws LdapException {
+        final LdapConnectionConfig config = new LdapConnectionConfig();
+        config.setLdapHost(ldapSettings.getUri().getHost());
+        config.setLdapPort(ldapSettings.getUri().getPort());
+        config.setUseSsl(ldapSettings.getUri().getScheme().startsWith("ldaps"));
+        config.setUseTls(ldapSettings.isUseStartTls());
+        if (ldapSettings.isTrustAllCertificates()) {
+            config.setTrustManagers(new TrustAllX509TrustManager());
+        }
+        config.setName(ldapSettings.getSystemUserName());
+        config.setCredentials(ldapSettings.getSystemPassword());
+        return ldapConnector.connect(config);
+    }
+
+    protected LdapEntry searchLdapUser(LdapNetworkConnection connection, String principal, LdapSettings ldapSettings) throws LdapException, CursorException {
+        return ldapConnector.search(connection,
+                ldapSettings.getSearchBase(),
+                ldapSettings.getSearchPattern(),
+                ldapSettings.getDisplayNameAttribute(),
+                principal,
+                ldapSettings.isActiveDirectory(),
+                ldapSettings.getGroupSearchBase(),
+                ldapSettings.getGroupIdAttribute(),
+                ldapSettings.getGroupSearchPattern());
+    }
+
     public boolean isEnabled() {
         final LdapSettings ldapSettings = ldapSettingsService.load();
         return ldapSettings != null && ldapSettings.isEnabled();
     }
 
+    public void syncLdapUser(String principal) {
```

Comment:
The method signature should allow users to at least verify if the synchronization was successful, so instead of returning`void`, it should return a boolean value (with `true` == successful, `false` == unsuccessful).

Alternatively return the `User` instance which was synced and mark the method as `@Nullable`.

Reply:
Completely agree.